pkg/component/shared/nginx_ingress.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shared
+
+import (
+	"github.com/Masterminds/semver"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/gardener/pkg/component"
+	"github.com/gardener/gardener/pkg/component/nginxingress"
+	"github.com/gardener/gardener/pkg/utils/images"
+	"github.com/gardener/gardener/pkg/utils/imagevector"
+)
+
+// NewNginxIngress returns a deployer for nginx-ingress-controller.
+func NewNginxIngress(
+	c client.Client,
+	namespaceName string,
+	targetNamespace string,
+	imageVector imagevector.ImageVector,
+	kubernetesVersion *semver.Version,
+	config map[string]string,
+	loadBalancerAnnotations map[string]string,
+	loadBalancerSourceRanges []string,
```

Comment:
This argument seems to be unused?

Reply:
It's used in case of Shoot.

Reply:
What do you suggest to pass directly `nginxingress.Values` as an argument?

The arguments that you mentioned are passed directly to the `nginxingress.Values`

IIANM some of these arguments are used by the botanist - [ref](https://github.com/gardener/gardener/blob/master/pkg/operation/botanist/nginxingress.go#L46-L54)

Reply:
The argument is currently missing in `nginxingress.Values`, see [here](https://github.com/gardener/gardener/blob/351d72b4bb9ab4e9fc8e0907dfa824c8cec52011/pkg/component/shared/nginx_ingress.go#L57-L68)

Reply:
you are right, thank you